packages/ember/src/index.ts

```diff
@@ -0,0 +1,43 @@
+export function setupAppsignal(appsignal: any, Ember = (window as any).Ember) {
+  const prevHandler = Ember.onerror
+
+  if (!appsignal) {
+    throw new Error("No instance of Appsignal supplied.")
+  }
+
+  if (!Ember) {
+    throw new Error("No instance of Ember found.")
+  }
+
+  Ember.onerror = function(error: Error): void {
+    const span = appsignal.createSpan((span: any) =>
+      span.setAction("Ember.onerror").setError(error)
+    )
+
+    appsignal.send(span)
+
+    if (prevHandler && typeof prevHandler === "function") {
+      prevHandler.apply(this, arguments as any)
+    } else if (Ember.testing) {
+      throw error
+    }
+  }
+
+  Ember.RSVP.on("error", function(reason: any): void {
+    const span = appsignal.createSpan((span: any) =>
+      span.setAction("Ember.RSVP.on")
+    )
+
+    if (reason instanceof Error) {
+      span.setError(reason)
+    } else {
+      span.setError({
+        name: "UnhandledPromiseRejectionError",
```

Comment:
Not sure what this error name comes from? Is it always a promise rejection?

Reply:
It's always an unhandled promise rejection, this is a listener that is fired when Ember's internal promise implementation throws an unhandled exception.

Reply:
ah maybe that's a good thing to add as a code comment?

Reply:
Fixed in d0a0024.